src/lcals/FIRST_MIN-Cuda.cpp

```diff
@@ -75,32 +76,45 @@ void FIRST_MIN::runCudaVariantImpl(VariantID vid)
 
     FIRST_MIN_DATA_SETUP_CUDA;
 
-    startTimer();
-    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+    const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
+    MyMinLoc* mymin_block = new MyMinLoc[grid_size]; //per-block min value
 
-       FIRST_MIN_MINLOC_INIT;
+    MyMinLoc* dminloc;
+    cudaErrchk( cudaMalloc( (void**)&dminloc, 
+                            grid_size * sizeof(MyMinLoc) ) );
 
-       MyMinLoc* dminloc;
-       cudaErrchk( cudaMalloc( (void**)&dminloc, sizeof(MyMinLoc) ) );
-       cudaErrchk( cudaMemcpy( dminloc, &mymin, sizeof(MyMinLoc),
-                               cudaMemcpyHostToDevice ) );
-
-       const size_t grid_size = RAJA_DIVIDE_CEILING_INT(iend, block_size);
-       first_min<block_size><<<grid_size, block_size,
-                   sizeof(MyMinLoc)*block_size>>>( x,
-                                                   dminloc,
-                                                   iend );
-       cudaErrchk( cudaGetLastError() );
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
 
-       cudaErrchk( cudaMemcpy( &mymin, dminloc, sizeof(MyMinLoc),
-                               cudaMemcpyDeviceToHost ) );
-       m_minloc = RAJA_MAX(m_minloc, mymin.loc);
+      FIRST_MIN_MINLOC_INIT;
 
-       cudaErrchk( cudaFree( dminloc ) );
+      for (Index_type i = 0; i < static_cast<Index_type>(grid_size); i++) {
+        mymin_block[i] = mymin;
+      }
+      cudaErrchk( cudaMemcpy( dminloc, mymin_block, grid_size*sizeof(MyMinLoc),
+                              cudaMemcpyHostToDevice ) );      
```

Comment:
We can get rid of this, we overwrite the memory each time.

Reply:
All of it, the for-loop and the memcpy?  Isn't that representative of the RAJA CUDA variant, here?

https://github.com/LLNL/RAJAPerf/blob/develop/src/lcals/FIRST_MIN-Cuda.cpp#L113

Reply:
Nope, the RAJA cuda/hip variants don't do any any of this kind of setting memory or copying memory.

Reply:
Because we account for the initial value for the reduction during finalization?

Reply:
We originally had a RAJA_MAX here because mymin could have an initial value so we check against that:

https://github.com/LLNL/RAJAPerf/blob/develop/src/lcals/FIRST_MIN.hpp#L51

Since we got rid those maxes having an init value is redundant. Are we OK not having an init value here?

Reply:
The RAJA does have an init value that gets captured by the lambda

Reply:
@CRobeck after conferring with @MrBurmark, we agreed that this follows the memory setting/motion behavior of the RAJA variants the best. If this were a functionality test, not a performance comparison we would do it differently.

Reply:
Sounds good.